pkg/kubelet/cm/topologymanager/policy_restricted.go

```diff
@@ -40,7 +45,8 @@ func (p *restrictedPolicy) canAdmitPodResult(hint *TopologyHint) bool {
 
 func (p *restrictedPolicy) Merge(providersHints []map[string][]TopologyHint) (TopologyHint, bool) {
 	filteredHints := filterProvidersHints(providersHints)
-	hint := mergeFilteredHints(p.numaNodes, filteredHints)
-	admit := p.canAdmitPodResult(&hint)
-	return hint, admit
+	merger := NewHintMerger(p.numaInfo, filteredHints, PolicyBestEffort, p.opts)
```

Comment:
Should this pass `PolicyRestricted)`?

Reply:
good catch, copy paste type of error, `p.Name()` sounds good. Let me fix that.